jax/_src/numpy/ufuncs.py

```diff
@@ -1603,9 +1603,10 @@ def arctan2(x1: ArrayLike, x2: ArrayLike, /) -> Array:
 
     To reconstruct the input angle, we might be tempted to use the identity
     :math:`\tan(\theta) = y / x`, and compute :math:`\theta = \tan^{-1}(y/x)`.
-    Unfortunately, this does not recover the input angle:
+    Unfortunately, this does not recover the input angle -- and the result will
+    generally depend on the details of your system's floating point rounding error:
```

Comment:
In general, `arctan(y/x)` does not cover the input angle because the sign information of `x` and `y` is mangled due to the division. That is, since `y/x == -y/(-x)`, `-y/x == y/(-x)`, etc then it would be impossible to recover the correct angle using solely the value of `y/x`.

I guess it also depends on the system's floating point properties but it will be noticeable only for corner cases while the actual reason for the issue lies in pure math.

Reply:
Yes - that's the point we're trying to make in this docstring. Do you have suggestions for how we could say it more clearly?

Reply:
This statement "the result will generally depend on the details of your system's floating point rounding error" is generally true for any floating-point calculation and therefor superfluous by itself.

What is the reason for making the point of the result being system dependent? Is this about numerical errors that occur/amplify by the division operation? If so, perhaps say that `arctan2(y, x)` is numerically more stable/accurate than `arctan(y/x)`.

Reply:
The direct issue is that `cos(jnp.pi / 2)` sometimes is slightly smaller than zero, and sometimes is slightly larger, and so the `arctan` result may toggle between `pi / 2` and `-pi / 2` – which is much more surprsing than a typical float rounding error.

Reply:
I think maybe we should just remove the new text. It was meant to comment on a subtlety of the output, but it seems the comment on the subtlety is overshadowing the main point the doc is trying to convey.

Reply:
#31197 removes it.

Reply:
Yeah I thought of that, but I think using multiples of pi/2 makes things clearer.